can_tools/scrapers/official/WA/wa_vaccine.py

```diff
@@ -5,65 +5,154 @@
 import re
 
 from can_tools.scrapers import variables
-from can_tools.scrapers.official.base import StateDashboard, MicrosoftBIDashboard
+from can_tools.scrapers.official.base import MicrosoftBIDashboard
 from can_tools.scrapers.puppet import with_page
 from can_tools.scrapers.util import flatten_dict
 
 
-class WashingtonVaccine(StateDashboard):
+class WashingtonVaccine(MicrosoftBIDashboard):
     has_location = False
     location_type = "county"
     state_fips = int(us.states.lookup("Washington").fips)
     source = "https://www.doh.wa.gov/Emergencies/COVID19/DataDashboard"
+    powerbi_url = "https://wabi-us-gov-virginia-api.analysis.usgovcloudapi.net"
     source_name = "Washington State Department of Health"
 
     variables = {
-        "People Initiating Vaccination": variables.INITIATING_VACCINATIONS_ALL,
-        "People Fully Vaccinated": variables.FULLY_VACCINATED_ALL,
+        "initiated": variables.INITIATING_VACCINATIONS_ALL,
+        "completed": variables.FULLY_VACCINATED_ALL,
     }
 
     async def _get_from_browser(self):
+        """use an async function to wait until the javascript has loaded to extract the iframe url.
+
+        The page is protected by a "no-javascript" blocker, so we cannot parse the html directly.
+        """
         async with with_page(headless=True) as page:
-            await page.goto(
-                "https://www.doh.wa.gov/Emergencies/COVID19/DataDashboard#downloads"
-            )
-            sel = "#accVaccinationsTbl table"
-            table_div = await page.waitForSelector(sel)
-            print("found table!")
-            table = await page.J(sel)
-            return await page.evaluate(" x => x.outerHTML", table)
+            await page.goto(self.source)
+            sel = "#dnn_ctr34282_HtmlModule_lblContent div"
+            iframe_div = await page.waitForSelector(sel)
+            iframe = await page.J(sel)
+            iframe = await page.evaluate(" x => x.outerHTML", iframe)
+            return {"src": re.findall(r'pbi-resize-src="(.*?)"', iframe)[0]}
 
-    def fetch(self):
+    def get_dashboard_iframe(self):
         return asyncio.get_event_loop().run_until_complete(self._get_from_browser())
 
-    def normalize(self, data: str) -> pd.DataFrame:
+    def construct_body(self, resource_key, ds_id, model_id, report_id):
+        "Build body request"
```

Comment:
nit: I think it's convention to use `"""` even for one-line docstrings. https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

> Triple quotes are used even though the string fits on one line. This makes it easy to later expand it.